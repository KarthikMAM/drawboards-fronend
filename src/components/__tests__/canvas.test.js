import React from 'react'
import { mount } from 'enzyme'

import Canvas from '../Canvas'

describe(Canvas, () => {
  it('should mount without any problems', () => {
    const wrapper = mount(<Canvas background='hello.png' />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should show canvas after the image loads', () => {
    const wrapper = mount(<Canvas background='hello.png' />)

    wrapper.find('img').simulate('load')

    expect(wrapper).toMatchSnapshot()
  })

  it('should set height and width of canvas based on the current state', () => {
    const wrapper = mount(<Canvas background='hello.png' />)

    wrapper.find('img').simulate('load')
    wrapper.setState({
      height: 400,
      width: 300,
    })

    expect(wrapper).toMatchSnapshot()
  })

  describe('draw', () => {
    it('should draw the required shapes on the canvas without creashing', () => {
      expect(() => {
        const wrapper = mount(<Canvas background='hello.png' />)

        wrapper.find('img').simulate('load')
        wrapper.setState({
          height: 400,
          width: 300,
        })
        wrapper.setProps({
          shapes: require('./__mocks__/canvasShapes/sample-1.json').shapes,
        })
      }).not.toThrowError()
    })
  })
})
