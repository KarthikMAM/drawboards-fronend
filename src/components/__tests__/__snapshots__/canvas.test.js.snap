// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canvas should mount without any problems 1`] = `
<div
  className="Container"
>
  <img
    alt="background"
    className="Background"
    onLoad={[Function]}
    src="https://test.cloud.com/hello.png"
  />
</div>
`;

exports[`Canvas should set height and width of canvas based on the current state 1`] = `
<div
  className="Container"
>
  <img
    alt="background"
    className="Background"
    onLoad={[Function]}
    src="https://test.cloud.com/hello.png"
  />
  <canvas
    className="Canvas"
    height={400}
    width={300}
  />
</div>
`;

exports[`Canvas should show canvas after the image loads 1`] = `
<div
  className="Container"
>
  <img
    alt="background"
    className="Background"
    onLoad={[Function]}
    src="https://test.cloud.com/hello.png"
  />
  <canvas
    className="Canvas"
    height={0}
    width={0}
  />
</div>
`;
